File: src/domain/Lotto.java
package domain;

import java.util.List;

//로또 번호 생성 및 저장, 결과
public interface Lotto {
    int MINNUM = 1;
    int MAXNUM = 45;
    int SIZE = 6;
    long PRICE = 1000;

    void createNum();

    void createNum(List<Integer> numbers);

    List<Integer> getNumbers();
}

File: src/main/scala/domain/Lotto.scala
package domain
import scala.util.Random.shuffle
import domain.Constants._
import scala.collection.mutable.ListBuffer

class Lotto(_numbers: List[Int]) {
  def numbers: List[Int] = _numbers
}

object Lotto {
  def apply() : Lotto = new Lotto(shuffle((1 to 45).toList).take(LOTTO_SIZE).sorted)

  def makeRandomLottos(numberOfLottos: Int): List[Lotto] = {
    val lottos = new ListBuffer[Lotto]()
    for (_ <- 1 to numberOfLottos) lottos += Lotto()
    lottos.toList
  }
}
